documentation/modules/ROOT/pages/02-architecture.adoc

```diff
@@ -1,104 +1,255 @@
+The capture should be performed regardless of any potential operational disruptions affecting dependent services (e.g., invoicing).
+
+In such scenarios, transactional applications typically return an error, which prevents any data from being changed, and causes the loss of real-world intent of end-users.
+This results in an adverse user experience and a deviation from the genuine business process, potentially leading to customer dissatisfaction.
+
+=== Solution Breakdown
+
+The concept of employing the _Event Mesh_ as a central, reliable hub for dispatching commands, as events, lies at the heart of this solution.
+This approach aligns closely with the Command Query Responsibility Segregation (_CQRS_) pattern, which distinctly categorizes commands and queries.
+Commands, in this context, are modeled as asynchronous events, designed to undergo processing by the _Event Mesh_.
+On the other hand, queries are synchronous operations, safe to retry, ensuring no loss of data integrity due to transient errors.
+
+The primary responsibility of the _Event Mesh_ is twofold.
+Firstly, it persists the incoming events, thereby maintaining a record of changes in the system's state.
+Secondly, it routes these events to their respective endpoints, ensuring that the appropriate microservices are notified and can subsequently update their internal states based on the event data.
+
+The mesh's inherent resilience is further bolstered by its exponential backoff strategy, which it employs when encountering operational failures.
+This mechanism ensures that the system retries the operation until it succeeds, thus mitigating the risk of data loss or system disruption due to transient issues.
+
+By integrating the _Event Mesh_ into the system architecture, several architectural benefits are achieved:
+
+* **Decomposition of the application into independently functioning services**:
+This approach facilitates a division of labor, with each service handling specific responsibilities.
+This not only enhances maintainability but also fosters scalability, as services can be independently scaled based on their demands.
 
-=== Using images 
+* **Improved business alignment**:
+By embracing an eventual consistency model, the _Event Mesh_ aligns closely with the inherent nature of most real-world business processes.
+Unlike traditional transactional systems that strive for immediate, irreversible consistency, the _Event Mesh_ allows for a more flexible and adaptive approach to data consistency.
+This results in better alignment with business requirements, as it supports scenarios where multiple services collaborate and synchronize their operations, making the whole state eventually consistent, without the constraint of strict, synchronous consistency.
 
-image::red_hat_open-hybrid-cloud.png[width=30%]
+* **Avoidance of systemic issues**:
+The _Event Mesh_'s error-handling mechanism, through retries and event persistence, aids in minimizing the impact of failures on the end user.
```

Comment:
We also have "dead letter sink" that can be optionally configured. 

so, even if a "consumer" is unable to receive the event. It will be routed to the DLS.

Now, if the DLS is also a Knative Broker, the benefit is that all undelivered events can be consumed from there, using the standard Knative Broker and Trigger APIs.

Reply:
I was reluctant to introduce the DLS to not complicate the solution even further. Maybe I can put some hint about it, without introducing it fully?!?

Reply:
> I was reluctant to introduce the DLS to not complicate the solution even further.

I see your point. Maybe we can just hint to the concept, and link to the docs as "out of scope" on _this_ doc? I geuss that would be fine, and still "complete"